tests/fastai/test_fastai_autolog.py

```diff
@@ -177,20 +177,18 @@ def test_fastai_autolog_model_can_load_from_artifact(fastai_random_data_run):
 def fastai_random_data_run_with_callback(iris_data, fit_variant, manual_run, callback, patience):
     # pylint: disable=unused-argument
     mlflow.fastai.autolog()
-    callbacks = []
 
-    if callback == "early":
-        # min_delta is set as such to guarantee early stopping
-        callbacks.append(
-            lambda learn: EarlyStoppingCallback(learn, patience=patience, min_delta=MIN_DELTA)
-        )
+    model = fastai_model(iris_data)
 
-    model = fastai_model(iris_data, callback_fns=callbacks)
+    callbacks = []
+    if callback == "early":
+        callback = EarlyStoppingCallback(learn=model, patience=patience, min_delta=MIN_DELTA)
+        callbacks.append(callback)
 
     if fit_variant == "fit_one_cycle":
-        model.fit_one_cycle(NUM_EPOCHS)
+        model.fit_one_cycle(NUM_EPOCHS, callbacks=callbacks)
     else:
-        model.fit(NUM_EPOCHS)
+        model.fit(NUM_EPOCHS, callbacks=callbacks)
```

Comment:
@harupy These changes are necessary in order to pass the fast.ai autologging tests now that `try_mlflow_log` throws in test mode. The fastai auotlogging call to `log_model` has been silently failing in our test environment because it attempts to pickle a lambda function (line 185) defined within the scope of another function; the version of pickle used by fast.ai doesn't support this.

As a workaround, instead of passing callbacks to the constructor of the fastai model (which requires that we use some sort of lambda or nested function in order to pass the model instance & patience value to the callback), we pass the callback to the `fit()` call instead.

Reply:
Nice fix!